dev/domain-controller/new-agent-vm/Install-PCoIPAgent.sh

```diff
@@ -0,0 +1,210 @@
+#!/bin/bash
+
+install_driver()
+{
+    # Install kernel-source/kernel-devel and gcc
+    sudo yum -y install kernel-source kernel-devel gcc
+    FILE_NAME='NVIDIA-Linux-x86_64-367.106-grid.run'
+    FILE_LOCATION='/root/'$FILE_NAME
+    # Download Driver first
+    echo "Downloading and Installing Nvidia driver"
+    sudo wget --retry-connrefused --tries=3 --waitretry=5 -O $FILE_LOCATION https://binarystore.blob.core.windows.net/thirdparty/nvidia/$FILE_NAME
+    exitCode=$?
+    if [ $exitCode -ne 0 ]
+    then
+        echo "failed to download Nvidia driver."
+        # let's define exit code 103 for this case
+        exit 103
+    fi
+    # Change file permission
+    sudo chmod 744 $FILE_LOCATION
+    # run installer
+    sudo $FILE_LOCATION -Z -X -s
+    exitCode=$?
+    
+    if [ $exitCode -eq 0 ]
+    then
+        echo "Driver is installed successfully"
+    else
+        echo "failed to install Nvidia driver. Will create a script to install driver when machine boots up"
+        file_path=/root/install_driver.sh
+        cat <<EOF >$file_path
+#!/bin/bash
+if [ -e /root/.first_boot ]
+then
+    if [ -e /root/.second_boot ]
+    then
+        exit 0
+    else
+        touch /root/.second_boot
+        sudo $FILE_LOCATION -Z -X -s        
+    fi
+else
+    touch /root/.first_boot
+    sudo $FILE_LOCATION -Z -X -s
+    (sleep 2;  sudo shutdown -f -r +0)&
+fi
+EOF
+        sudo chmod 744 $file_path
+        (crontab -l 2>/dev/null; echo "@reboot $file_path") | crontab -
+        # let's define exit code 104 for this case
+        #exit 104
+    fi
+}
+
+# the first argument is the Registration Code of PCoIP agent
+REGISTRATION_CODE=$1
+# the second argument is the agent type
+AGENT_TYPE=$2
+# the third argument is the VM name
+VM_NAME=$3
+# the forth argument is the domain name
+DOMAIN_NAME=$4
+# the fifith argument is the username
+USERNAME=$5
+# the sixth argument is the password
+PASSWORD=$6
+# the seventh argument is the domain group to join
+GROUP=$7
```

Comment:
It's not joining a domain group though, is that correct?

Reply:
Yes, I thought we could join the domain group but it seems it would require a lot of work to do it.

Reply:
From the document to install PCoIP Agent on REPL
https://techsupport.teradici.com/ics/support/kbanswer.asp?deptID=15164&task=knowledge&questionID=3147

Reply:
Yeah, sorry I should have been thinking about the 'done' criteria more when we were talking a few days ago.  This site seems to have a binary for adtool - from 2009??? https://directory.fsf.org/wiki/Adtool